extensions/wikia/AdEngine/js/PageFairDetection.js

```diff
@@ -0,0 +1,71 @@
+define('ext.wikia.adEngine.pageFairDetection', [
+	'ext.wikia.adEngine.adContext',
+	'ext.wikia.adEngine.utils.scriptLoader',
+	'wikia.browserDetect',
+	'wikia.document',
+	'wikia.log',
+	'wikia.window'
+], function (adContext, scriptLoader, browserDetect, doc, log, win) {
+	'use strict';
+
+	var context = null,
+		logGroup = 'ext.wikia.adEngine.pageFairDetection';
+
+	function getContext() {
```

Comment:
As you said, for Mercury context is not fully build, @bognix suggested a different solution - inject context by init function. What do You think about this idea @drets, @fraszczakszymon?